api/datasets_test.go

```diff
@@ -461,21 +465,18 @@ func TestDatasetGet(t *testing.T) {
 	}
 	run.SaveDataset(&ds, "testdata/cities/data.csv")
 
-	// TODO(dustmop): Would be nice to have a "fuzzy" json comparison, either as a third-party
-	// library, or something we develop, that would make tests like this easier to read and
-	// reason about. There's certain values in this json that we really care about (format),
-	// and then there's some we don't care about at all (signature).
-
 	actualStatusCode, actualBody := APICall("/get/peer/test_ds", dsHandler.GetHandler)
-	expectBody := `{"data":{"peername":"peer","name":"test_ds","path":"/map/QmRvuuXS4cPeZnxMMaXKaqMjEa3tunqZd6r3c4HEH2h3KN","dataset":{"bodyPath":"/map/QmVYgdpvgnq3FABZFVWUgxr7UCwNSRJz97vBU9YX5g5pQ4","commit":{"author":{"id":"QmZePf5LeXow3RW5U1AgEiNbW46YnRGhZ7HPvm1UmPFPwt"},"message":"created dataset from data.csv","path":"/map/QmbsySjGEJUqFWNfdiPTPtxyba6wf86TmbQe1nifpZCRH6","qri":"cm:0","signature":"TzHbqw7oRcqoGYhTraiJ9fSGzUUoHA653jNYGsBgbkUbBUkTh/ymTHZSfhwIKQlAqfwiOqB2rbmA4hu2MAYVxNRLfPlUYAr38juyTosI6cljYdzLfNk2L788iFSQcYnJ9CgiHYQlhmpUjh3irFY1nDIuaiPL1vDxH5yGsuI3qiD9DVvu2P6f6GSGMkQzpSv3jDWQbqo5LyyU7gaJBIaJW6Q9vKSB/mRRjDfHtOP2pMH/pf17q35+HaGHd+XEg/6a6X3IWwjsRros029lH6SDCbUaPXB6H3Cy5gRLoZp7K3mU026JucogrVHqRsZmVCx+vaVJ/MCpQhfYg6F8m8z2fA==","timestamp":"2001-01-01T01:01:01.000000001Z","title":"created dataset from data.csv"},"meta":{"qri":"md:0","title":"title one"},"name":"test_ds","path":"/map/QmRvuuXS4cPeZnxMMaXKaqMjEa3tunqZd6r3c4HEH2h3KN","peername":"peer","qri":"ds:0","structure":{"checksum":"QmVYgdpvgnq3FABZFVWUgxr7UCwNSRJz97vBU9YX5g5pQ4","depth":2,"entries":5,"format":"csv","formatConfig":{"headerRow":true,"lazyQuotes":true},"length":154,"qri":"st:0","schema":{"items":{"items":[{"title":"city","type":"string"},{"title":"pop","type":"integer"},{"title":"avg_age","type":"number"},{"title":"in_usa","type":"boolean"}],"type":"array"},"type":"array"}}},"published":false},"meta":{"code":200}}`
 	assertStatusCode(t, "get dataset", actualStatusCode, 200)
-	if diff := cmp.Diff(expectBody, actualBody); diff != "" {
+	expect := dstest.LoadGoldenFile(t, "testdata/expect/TestGetDataset.test_ds.json")
```

Comment:
>  However, if somehow the api handler were to break and not return the meta.title or structure.format, nothing in the buggy PR would reveal that. Yes, the hash would change, but that's opaque just as abide's binary files were. It would be invisible in the diff.

But... the golden file would change, right? and because it's in plaintext we can actually _review_ the changes to the file, which includes not just hashes, but the entire dataset response.

I can see the upside of an API that lets us fuzz-up golden files. I think another way to do this would be to write functions that remove parts of the dataset we don't want to compare to and write the result of calling those functions to golden files.

> it appears that this test could also benefit by combining boilerplate. Is there ever going to be a case where we want to CompareDatasets but not UpdateGoldenFileIfEnvVarSet?

There are cases where we're using them  separately, but I love the idea of a convenience function. I'm going to put it in detest because I'm tired of copying shit all over the place for tests